pykechain/models/partset.py

```diff
@@ -11,7 +11,10 @@ class PartSet(object):
 
     def __init__(self, parts):
         """Construct a PartSet from a part iterable."""
-        self._parts = list(parts)
+        if parts:
+            self._parts = list(parts)
```

Comment:
i check the case that parts are provided as None. If the list of parts are provided as [], the self._parts = list([]), which is a list with no content.